database/insertData/08_template_filters.js

```diff
@@ -0,0 +1,343 @@
+/*
+Template Filters, joined to all templates
+*/
+exports.queries = [
+  // "Approved Applications" - APPLY
+  `mutation addFilter {
+    createFilter(
+      input: {
+        filter: {
+          icon: "check circle"
+          code: "approveApplications"
+          query: { outcome: "APPROVED" }
+          title: "Approved Applications"
+          userRole: APPLY
+          templateFilterJoinsUsingId: {
+            create: [
+              { templateToTemplateId: { connectByCode: { code: "Demo" } } }
+              {
+                templateToTemplateId: {
+                  connectByCode: { code: "OrgRegistration" }
+                }
+              }
+              {
+                templateToTemplateId: {
+                  connectByCode: { code: "UserRegistration" }
+                }
+              }
+              { templateToTemplateId: { connectByCode: { code: "ReviewTest" } } }
+              { templateToTemplateId: { connectByCode: { code: "DrugRegoGen" } } }
+              { templateToTemplateId: { connectByCode: { code: "OrgJoin" } } }
```

Comment:
So this is why you made `code` unique (above)?

Perhaps the "template_id" field in template_filter_join needs to be a "soft" reference to template_code, rather than a foreign key. That's how we've done it elsewhere (for template elements, for example).

And also, if the AVAILABLE (Active) version of a template changes, we'd have to update all the template_id references. So linking by "code" would be better, it just wouldn't be baked into the database.

But that's okay -- sometimes we want a hard-link to an exact template versions (e.g. for existing applications), whereas sometimes we want a "soft" link to a template code, which will always stay linked to the "AVAILABLE" version of the template.

So, conclusion? Perhaps change "template_id" field in "template_filter_join" to "template_code" and not make it a foreign key.

Reply:
https://github.com/openmsupply/application-manager-server/pull/356#discussion_r632177974